test/integration/deployment/util.go

```diff
@@ -221,12 +222,33 @@ func (d *deploymentTester) markAllPodsReady() {
 			d.t.Logf("failed to list Deployment pods, will retry later: %v", err)
 			return false, nil
 		}
-		if len(pods.Items) != int(*d.deployment.Spec.Replicas) {
-			d.t.Logf("%d/%d of deployment pods are created", len(pods.Items), *d.deployment.Spec.Replicas)
+		rss, err := d.c.ExtensionsV1beta1().ReplicaSets(ns).List(metav1.ListOptions{LabelSelector: selector.String()})
```

Comment:
Suggest adding a function `func (d *deploymentTester) listPods()` to replace the List() call in L220. This way the code would be much cleaner. 

In `listPods()`, use getNewReplicaSet() to get new RS of the deployment, and only return the pods with controllerRef points to that new RS.